model-optimizer/extensions/middle/MoveConstToLoopBody.py

```diff
@@ -0,0 +1,51 @@
+"""
+ Copyright (C) 2017-2021 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import numpy as np
+
+from extensions.ops.loop import Loop
+from mo.middle.replacement import MiddleReplacementPattern
+from mo.graph.graph import Graph, Node
+from mo.ops.const import Const
+
+
+class MoveConstToLoopBody(MiddleReplacementPattern):
+    """
+    """
+    enabled = True
+
+    def run_after(self):
+        from extensions.middle.PartialInfer import PartialInfer
+        return [PartialInfer]
+
+    def run_before(self):
+        from extensions.middle.ApplyPermutations import ApplyPermutation
+        return [ApplyPermutation]
+
+    def find_and_replace_pattern(self, graph: Graph):
+        cleanup_called_once = False
+        # walk through all Loop nodes and find Const inputs
+        for loop_node in graph.get_op_nodes(op='Loop'):
+            if not cleanup_called_once:
+                graph.clean_up()
+                cleanup_called_once = True
+            Loop.pull_constant_inputs_into_body(loop_node)
+            Loop.normalize_input_output_ports(loop_node)
+            Loop.infer(loop_node)
```

Comment:
it needs additional infer in case of constant folding after pulling constant from the main graph.